app/src/main/java/app/revanced/integrations/syncforreddit/internal/RedditVideoPlaylistParser.kt

```diff
@@ -0,0 +1,51 @@
+package app.revanced.integrations.syncforreddit.internal
+
+import org.w3c.dom.Element
+import java.io.ByteArrayInputStream
+import javax.xml.parsers.DocumentBuilderFactory
+
+object RedditVideoPlaylistParser {
```

Comment:
You can move this to the Java class. No need to use Kotlin, because Kotlin ships its stdlib and when the app you patch has a different version that can cause conflicts.